scripts/utils.js

```diff
@@ -85,13 +85,63 @@ function buildTagsBlock() {
   }
 }
 
+function getImageCaption(picture) {
+  // Check if the parent element has a caption
+  const parentEl = picture.parentNode;
+  let caption = parentEl.querySelector('em');
+  if (caption) return caption;
+
+  // If the parent element doesn't have a caption, check if the next sibling does
+  const parentSiblingEl = parentEl.nextElementSibling;
+  caption = parentSiblingEl
+    && !parentSiblingEl.querySelector('picture')
+    && parentSiblingEl.firstChild.tagName === 'EM'
+    ? parentSiblingEl.querySelector('em')
+    : undefined;
```

Comment:
nit: a bit complex for a ternary, maybe an early return? Also caption can equal the last conditional, no need to query.
```js
? parentSiblingEl.firstChild
```